cloudinary_cli/modules/clone.py

```diff
@@ -0,0 +1,125 @@
+from cloudinary_cli.utils.config_utils import load_config, \
+     refresh_cloudinary_config, verify_cloudinary_url
+from cloudinary_cli.defaults import logger
+from cloudinary_cli.core.search import execute_single_request, \
+     handle_auto_pagination
+import os
+
+DEFAULT_MAX_RESULTS = 500
+
+
+@command("clone",
+         short_help="""Clone assets from one account to another.""",
+         help="""
+\b
+Clone assets from one environment to another with/without tags and context (structured metadata is not currently supported).
+Source will be your `CLOUDINARY_URL` environemnt variable but you also can specify a different source using `-c/-C` option.
+Cloning restricted assets is also not supported currently.
+Format: cld clone -t/-T <target_environemnt> <command options>
+You need to specify the target cloud via `-t` or `-T` (not both)
+e.g. cld clone -t cloudinary://<api_key>:<api_secret>@<cloudname> -f tags,context -O
+""")
+@option("-T", "--target_saved",
+        help="Tell the CLI the target environemnt to run the command on by specifying a saved configuration - see `config` command.")
+@option("-t", "--target",
+        help="Tell the CLI the target environemnt to run the command on by specifying an environment variable.")
+@option("-A", "--auto_paginate", is_flag=True, default=False,
+        help="Auto-paginate Admin API calls.")
+@option("-F", "--force", is_flag=True,
+        help="Skip confirmation.")
+@option("-O", "--overwrite", is_flag=True, default=False,
+        help="Skip confirmation.")
+@option("-w", "--concurrent_workers", type=int, default=30,
+        help="Specify the number of concurrent network threads.")
+@option("-f", "--fields", multiple=True,
+        help="Specify whether to copy tags and context.")
+@option("-se", "--search_exp", default="",
+        help="Define a search expression.")
+@option("--async", "async_", is_flag=True, default=False,
+        help="Generate asynchronously.")
+@option("-nu", "--notification_url",
+        help="Webhook notification URL.")
+def clone(target_saved, target, auto_paginate, force,
+          overwrite, concurrent_workers, fields, search_exp,
+          async_, notification_url):
+    if bool(target) == bool(target_saved):
+        print_help_and_exit()
+
+    base_cloudname_url = os.environ.get('CLOUDINARY_URL')
+    base_cloudname = cloudinary.config().cloud_name
+    if target:
+        verify_cloudinary_url(target)
+    elif target_saved:
+        config = load_config()
+        if target_saved not in config:
+            logger.error(f"Config {target_saved} does not exist")
+            return False
+        else:
+            refresh_config(target_saved=target_saved)
+    target_cloudname = cloudinary.config().cloud_name
+    if base_cloudname == target_cloudname:
+        logger.info("Target environment cannot be the "
+                    "same as source environment.")
+        return True
+    refresh_config(base_cloudname_url)
```

Comment:
Instead of loading target, and then resetting, you can just create a new instance of `cloudinary.Config()` class and use it.